citations-tool/tool/src/java/org/sakaiproject/citation/tool/CitationHelperAction.java

```diff
@@ -4072,42 +4138,36 @@ public void doReviseCitation (RunData data)
 		{
 			String citationId = (String) state.getAttribute(CitationHelper.CITATION_EDIT_ID);
 			int location = (Integer) state.getAttribute("location");
+			Citation citation;
 			if(citationId != null)
 			{
-				try
-	            {
-		            // if it's a unnested citation
-		            if (location==0) {
-			            String citationCollectionId = (String) state.getAttribute("citation.citation_collection_id");
-			            collection = getCitationService().getUnnestedCitationCollection(citationCollectionId);
-		            }
-
-		            Citation citation = collection.getCitation(citationId);
-
-		            String schemaId = params.getString("type");
-		            Schema schema = getCitationService().getSchema(schemaId);
-		            citation.setSchema(schema);
-
-		    		updateCitationFromParams(citation, params);
-
-		       		// add citation to current collection
-		    		collection.saveCitation(citation);
-		        }
-	            catch (IdUnusedException e)
-	            {
-		            // TODO add alert and log error
-	            }
-
-	       		if (location==0){
-			        getCitationService().save(collection);
-		        }
+				try {
+					// if it's a unnested citation
+					if (location == 0) {
+						String citationCollectionId = (String) state.getAttribute("citation.citation_collection_id");
+						collection = getCitationService().getUnnestedCitationCollection(citationCollectionId);
+					}
+					citation = collection.getCitation(citationId);
+				}
+				catch (IdUnusedException e){
+					citation = (Citation)state.getAttribute(CitationHelper.CITATION_EDIT_ITEM);
+				}
+				String schemaId = params.getString("type");
+				Schema schema = getCitationService().getSchema(schemaId);
+				citation.setSchema(schema);
+				updateCitationFromParams(citation, params);
+				if(!collection.contains(citation)){
+					collection.add(citation);
+				}
+			}
+			// add citation to current collection
+			if (location==0){
+				getCitationService().save(collection);
 			}
```

Comment:
Added  if(!collection.contains(citation)){
					collection.add(citation);
         	} 
and assigning citation in catch block